grpc/server/src/main/java/io/helidon/grpc/server/ServiceDescriptor.java

```diff
@@ -719,6 +738,13 @@ private String getPackageName() {
             return "".equals(pkg) ? proto.getPackage() : pkg;
         }
 
+        /**
+         * Returns the service name prefixed with package directive if one exists.
+         */
+        private String getFullName() {
```

Comment:
I see, you need it within the `Builder` as well...

Maybe that's the only place you should have this code, and the outer `ServiceDescriptor` should simply have a `fullName` field that will be initialized by the `Builder`, along with other fields?

Reply:
Can you please check if the new change I pushed is the approach you are envisioning? I maintained ServiceDescriptor.getFullName() method as `protected` rather than `public` is it is only used within the package.